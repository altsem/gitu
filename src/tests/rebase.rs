use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "checkout", "-b", "other-branch"]);
    run(ctx.dir.path(), &["git", "checkout", "main"]);
    commit(ctx.dir.path(), "new-file", "hello");
    run(ctx.dir.path(), &["git", "checkout", "other-branch"]);
    ctx
}

#[test]
fn rebase_menu() {
    snapshot!(setup(), "r");
}

#[test]
fn rebase_elsewhere_prompt() {
    snapshot!(setup(), "re");
}

#[test]
fn rebase_elsewhere() {
    snapshot!(setup(), "remain<enter>");
}
