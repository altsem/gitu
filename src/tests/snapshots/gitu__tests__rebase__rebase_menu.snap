---
source: src/tests/rebase.rs
expression: ctx.redact_buffer()
---
▌On branch other-branch                                                         |
                                                                                |
 Recent commits                                                                 |
 b66a0bf other-branch origin/main add initial-file                              |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
Rebase                  Arguments                                               |
a abort                 -a Autosquash (--autosquash)                            |
c continue              -A Autostash (--autostash)                              |
e onto elsewhere        -d Lie about committer date (--committer-date-is-author-|
q/<esc> Quit/Close      -i Interactive (--interactive)                          |
                        -k Keep empty commits (--keep-empty)                    |
                        -h Disable hooks (--no-verify)                          |
                        -p Preserve merges (--preserve-merges)                  |
styles_hash: c456a01c0e45182a
